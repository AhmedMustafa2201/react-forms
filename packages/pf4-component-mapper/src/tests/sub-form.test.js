import React from 'react';
import toJson from 'enzyme-to-json';
import SubForm from '../components/sub-form';
import { shallow } from 'enzyme';

describe('SubForm component', () => {
  const props = {
    title: 'cosiTitle',
    name: 'cosiName',
    fields: [],
    formOptions: {
      renderForm: () => <div>Here would be form</div>
    }
  };

  it('should render SubForm correctly', () => {
    const wrapper = shallow(<SubForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render SubForm with description correctly', () => {
    const propsDescription = { ...props, description: 'description here!' };
    const wrapper = shallow(<SubForm {...propsDescription} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render SubForm without title correctly', () => {
    const { name, fields, formOptions } = props;
    const propsWithoutTitle = { name, fields, formOptions };
    const wrapper = shallow(<SubForm {...propsWithoutTitle} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
