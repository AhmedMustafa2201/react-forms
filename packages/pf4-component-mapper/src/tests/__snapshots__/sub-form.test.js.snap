// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubForm component should render SubForm correctly 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <SubForm
      fields={Array []}
      name="cosiName"
      title="cosiTitle"
    />
  </ContextProvider>
</ContextProvider>
`;

exports[`SubForm component should render SubForm with description correctly 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <SubForm
      description="description here!"
      fields={Array []}
      name="cosiName"
      title="cosiTitle"
    />
  </ContextProvider>
</ContextProvider>
`;

exports[`SubForm component should render SubForm without title correctly 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": undefined,
      }
    }
  >
    <SubForm
      fields={Array []}
      name="cosiName"
    />
  </ContextProvider>
</ContextProvider>
`;
